akka-http-core/src/main/scala/akka/http/javadsl/ServerBuilder.scala

```diff
@@ -0,0 +1,81 @@
+package akka.http.javadsl
+
+import java.util.concurrent.CompletionStage
+
+import akka.actor.ClassicActorSystemProvider
+import akka.annotation.ApiMayChange
+import akka.event.LoggingAdapter
+import akka.http.impl.util.JavaMapping.Implicits._
+import akka.http.javadsl.model.{ HttpRequest, HttpResponse }
+import akka.http.javadsl.settings.ServerSettings
+import akka.http.scaladsl
+import akka.http.scaladsl.{ model => sm }
+import akka.japi.function.Function
+import akka.stream.javadsl.Flow
+
+import scala.compat.java8.FutureConverters._
+import scala.concurrent.ExecutionContext
+
+trait HandlerProvider {
```

Comment:
Smallest common denominator between akka-http-core and akka-http to allow passing a route directly into `bind` without needing to know about `Route` here.

Reply:
Perhaps clarify in the javadoc that `Route` is the typically used implementation?

Reply:
Done now also in the docs for `bind(HandlerProvider)`.